src/theory/arith/branch_and_bound.cpp

```diff
@@ -0,0 +1,143 @@
+    Node right = nm->mkNode(OR, ub, lb);
+    Node rawEq = nm->mkNode(EQUAL, var, mkRationalNode(nearest));
+    Node eq = Rewriter::rewrite(rawEq);
+    // Also preprocess it before we send it out. This is important since
+    // arithmetic may prefer eliminating equalities.
+    TrustNode teq;
+    if (Theory::theoryOf(eq) == THEORY_ARITH)
+    {
+      teq = d_ppre.ppRewriteEq(eq);
+      eq = teq.isNull() ? eq : teq.getNode();
+    }
+    Node literal = d_astate.getValuation().ensureLiteral(eq);
+    Trace("integers") << "eq: " << eq << "\nto: " << literal << std::endl;
+    d_im.requirePhase(literal, true);
+    Node l = nm->mkNode(OR, literal, right);
+    Trace("integers") << "l: " << l << std::endl;
+    if (proofsEnabled())
+    {
+      Node less = nm->mkNode(LT, var, mkRationalNode(nearest));
+      Node greater = nm->mkNode(GT, var, mkRationalNode(nearest));
+      // TODO (project #37): justify. Thread proofs through *ensureLiteral*.
+      Debug("integers::pf") << "less: " << less << std::endl;
+      Debug("integers::pf") << "greater: " << greater << std::endl;
+      Debug("integers::pf") << "literal: " << literal << std::endl;
+      Debug("integers::pf") << "eq: " << eq << std::endl;
+      Debug("integers::pf") << "rawEq: " << rawEq << std::endl;
+      Pf pfNotLit = d_pnm->mkAssume(literal.negate());
+      // rewrite notLiteral to notRawEq, using teq.
+      Pf pfNotRawEq =
+          literal == rawEq
+              ? pfNotLit
+              : d_pnm->mkNode(
+                    PfRule::MACRO_SR_PRED_TRANSFORM,
+                    {pfNotLit,
+                     teq.getGenerator()->getProofFor(teq.getProven())},
+                    {rawEq.negate()});
+      Pf pfBot = d_pnm->mkNode(
+          PfRule::CONTRA,
+          {d_pnm->mkNode(PfRule::ARITH_TRICHOTOMY,
+                         {d_pnm->mkAssume(less.negate()), pfNotRawEq},
+                         {greater}),
+           d_pnm->mkAssume(greater.negate())},
+          {});
+      std::vector<Node> assumptions = {
+          literal.negate(), less.negate(), greater.negate()};
+      // Proof of (not (and (not (= v i)) (not (< v i)) (not (> v i))))
+      Pf pfNotAnd = d_pnm->mkScope(pfBot, assumptions);
+      Pf pfL = d_pnm->mkNode(PfRule::MACRO_SR_PRED_TRANSFORM,
+                             {d_pnm->mkNode(PfRule::NOT_AND, {pfNotAnd}, {})},
+                             {l});
+      lem = d_pfGen->mkTrustNode(l, pfL);
+    }
+    else
+    {
+      lem = TrustNode::mkTrustLemma(l, nullptr);
+    }
+  }
+  else
+  {
+    Node ub = Rewriter::rewrite(nm->mkNode(LEQ, var, mkRationalNode(floor)));
+    Node lb = ub.notNode();
+    if (proofsEnabled())
+    {
+      lem =
+          d_pfGen->mkTrustNode(nm->mkNode(OR, ub, lb), PfRule::SPLIT, {}, {ub});
+    }
+    else
+    {
+      lem = TrustNode::mkTrustLemma(nm->mkNode(OR, ub, lb), nullptr);
+    }
+  }
+
+  Trace("integers") << "integers: branch & bound: " << lem << std::endl;
```

Comment:
What about the additional debug output [here](https://github.com/cvc5/cvc5/blob/master/src/theory/arith/theory_arith_private.cpp#L3554-L3577)?

Reply:
This uses an internal method `getSatValue`; I'm not transferring this part of the debugging.